### PR TITLE
sp - FEATURE: Admin table should not move entries after a toggle button is pressed

### LINKED ISSUE #15: FEATURE: Admin table should not move entries after a toggle button is pressed
### As a...

admin

### I can...

toggle a user's admin, driver, or rider without moving the entries in the table

### So that...

the user's table stays consistent

### Discussion

Here is what is currently happening:

User table before toggling "Driver" for Aasish Virjala:

Screenshot 2024-05-15 at 5 23 44 PM

User table after toggling "Driver" for Aasish Virjala:

Screenshot 2024-05-15 at 5 23 51 PM

This needs to be fixed so that the entries in the table stay consistent.

### Acceptance Criteria

- [ ] User table is sorted by some metric (you decide...)
- [ ] When a property of a user is toggled, they should remain in the same position in the table (instead of moving to the bottom)

### Implementation Todos

Frontend:
- [ ] manipulate the table so that it stays sorted by some sort of metric (maybe id?)

Testing:
- [ ] Front end tests pass and there is adequate coverage

### PR DESCRIPTION
**Pull request by spark686**
I sorted the users table based on id. It the ordering of the  not move after admin pressing driver toggle button. 

+  User table is sorted by some metric
+ When a property of a user is toggled, they should remain in the same position in the table (instead of moving to the bottom)

image

dokku deployment:
      https://final-spark686-sort.dokku-16.cs.ucsb.edu

Closes #15

### CODE REVIEW

**Review by bradygin — DISMISSED** [review]
LGTM! Just linked your issue so that it will be closed when this is merged

**Review by Hannahshakouri01 — DISMISSED** [review]
LGTM!

File: frontend/src/main/components/Users/UsersTable.js [hunk 1]
```diff
@@ -1,8 +1,17 @@
-
 import OurTable, { ButtonColumn } from "main/components/OurTable"
 import { useBackendMutation } from "main/utils/useBackend";
+import React, { useState } from 'react'; // Added missing imports
+
+
 
 export default function UsersTable({ users}) {
+    // const [sortedUsers, setSortedUsers] = useState(users);
```
Comment (pconrad) [inline frontend/src/main/components/Users/UsersTable.js]: Commented out code?  https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html#why-should-commented-out-code-be-removed

File: frontend/src/main/components/Users/UsersTable.js [hunk 2]
```diff
@@ -1,8 +1,17 @@
...
+    const [sort, ] = useState({ keyToSort: 'id'});
+
+    // useEffect(() => {
+    //     setSortedUsers(sortTableById(users, sort));
+    // }, [users, sort]);
```
Comment (pconrad) [inline frontend/src/main/components/Users/UsersTable.js]: https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html#why-should-commented-out-code-be-removed

File: frontend/src/main/components/Users/UsersTable.js [hunk 3]
```diff
@@ -61,12 +70,25 @@ export default function UsersTable({ users}) {
...
         cellToAxiosParamsToggleDriver,
         {},
-        ["/api/admin/users"]
+        ["/api/admin/users"],
     );
     // Stryker enable all 
 
+    // function sortArray(array) {
+    //     return [...array].sort((a, b) => a.id - b.id);
+    // }
```
Comment (pconrad) [inline frontend/src/main/components/Users/UsersTable.js]: https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html#why-should-commented-out-code-be-removed

File: frontend/src/main/components/Users/UsersTable.js [hunk 4]
```diff
@@ -113,7 +135,7 @@ export default function UsersTable({ users}) {
     //const columnsToDisplay = showButtons ? buttonColumn : columns;
```
Comment (pconrad) [inline frontend/src/main/components/Users/UsersTable.js]: https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html#why-should-commented-out-code-be-removed

File: frontend/src/tests/components/Users/UsersTable.test.js [hunk 5]
```diff
@@ -1,12 +1,26 @@
...
 import UsersTable from "main/components/Users/UsersTable"
+
 import { QueryClient, QueryClientProvider } from "react-query";
 
+const mockedNavigate = jest.fn();
+
+// jest.mock('react-router-dom', () => ({
+//     ...jest.requireActual('react-router-dom'),
+//     useNavigate: () => mockedNavigate
+// }));
```
Comment (pconrad) [inline frontend/src/tests/components/Users/UsersTable.test.js]: https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html#why-should-commented-out-code-be-removed

**Review by pconrad — CHANGES_REQUESTED** [review]
https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html#why-should-commented-out-code-be-removed

**Review by pconrad — APPROVED** [review]
LGTM... will test on dokku before merging

**Review by whuang602 — APPROVED** [review]
LGTM

**Pull request merged**